components/core/src/clp/FileReader.cpp

```diff
@@ -12,15 +12,27 @@
 using std::string;
 
 namespace clp {
+FileReader::FileReader(string const& path)
+        : m_file{fopen(path.c_str(), "rb")},
```

Comment:
is this the right way to intialize m_file?
I get a clang tidy warning if I put this line in the function implementation.